### ISSUE TITLE
Doesn't work in strict mode

### ISSUE THREAD

**Issue by kshutkin** [RubenVerborgh/promiscuous#37]
It is smallest Promise es6/Promises A+ implementation that I know about. I was using it successfully for some time but with new build tools that supports Scope hoisting (rollup, webpack 3) it doesn't play well. 

The problem is valueHandler function definition is not compatible with strict mode. I have to either prepend this lib directly to result file excluding from main bundle or to turn strict mode off.

**Comment by RubenVerborgh** [comment]
> The problem is valueHandler function definition is not compatible with strict mode

I don't see anything particularly wrong with that function. It can be hoisted without any problem.

I also tried the file in strict mode, and did not find any problems. Closing this for now until we can reproduce.

**Comment by railsbros-dirk** [comment]
We have currently the same problem. It doesn't affect all browsers though. We have the problem with:

* Mobile Safari <= 9
* Safari <=9
* Chrome <= 38
* Firefox <= 45
* Android <=4 and Android 6 for the build in "Internet" browser on LG phone

As you can see we have the problem with quite some visitors thus it should be easily reproducible:

<img width="1229" alt="screen shot 2017-08-09 at 14 02 04" src="https://user-images.githubusercontent.com/1948/29120720-ec9b529a-7d0b-11e7-9dd0-6c715f1909ac.png">

As already mentioned the problem is with the `valueHandler`:

```json
var promiscuous = createCommonjsModule(function (module) {
/**@license MIT-promiscuous-©Ruben Verborgh*/
(function (func, obj) {
  function is(type, item) { return (typeof item)[0] == type; }
  function Promise(callback, handler) {
    handler = function pendingHandler(resolved, rejected, value, queue, then, i) {
      queue = pendingHandler.q;
      if (resolved != is) {
        return Promise(function (resolve, reject) {
          queue.push({ p: this, r: resolve, j: reject, 1: resolved, 0: rejected });
        });
      }
      if (value && (is(func, value) | is(obj, value))) {
        try { then = value.then; }
        catch (reason) { rejected = 0; value = reason; }
      }
      if (is(func, then)) {
        function valueHandler(resolved) { // <----- We get the error on this line
          return function (value) { then && (then = 0, pendingHandler(is, resolved, value)); };
        }
        try { then.call(value, valueHandler(1), rejected = valueHandler(0)); }
        catch (reason) { rejected(reason); }
      }
```

Thanks for looking into this once more.

**Comment by RubenVerborgh** [comment]
Can you try the `strict-mode` branch?

Note that I'm trying blindly here, as I don't have any of the browsers to reproduce this.

**Comment by kshutkin** [comment]
for me it works

**Comment by RubenVerborgh** [comment]
Thanks! @railsbros-dirk, does it work for you?

**Comment by railsbros-dirk** [comment]
Thank you so much for you super fast response **and** action @RubenVerborgh. I just tested it on the above mentioned LG phone. Works perfectly. Thanks again:

![](https://media.giphy.com/media/3oz8xIsloV7zOmt81G/giphy.gif)

**Comment by RubenVerborgh** [comment]
You're welcome! Published as v0.7.1.
